pkg/controller/scyllacluster/resource_test.go

```diff
@@ -925,3 +927,227 @@ func TestStatefulSetForRack(t *testing.T) {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+				Labels: map[string]string{
+					"app":                          "scylla",
+					"app.kubernetes.io/name":       "scylla",
+					"app.kubernetes.io/managed-by": "scylla-operator",
+					"scylla/cluster":               "basic",
+					"scylla-operator.scylladb.com/scylla-ingress-type": ingressType,
+				},
+				Annotations: annotations,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "scylla.scylladb.com/v1",
+						Kind:               "ScyllaCluster",
+						Name:               "basic",
+						UID:                "the-uid",
+						Controller:         pointer.BoolPtr(true),
+						BlockOwnerDeletion: pointer.BoolPtr(true),
+					},
+				},
+			},
+			Spec: networkingv1.IngressSpec{
+				IngressClassName: pointer.String(ingressClassName),
+				Rules:            rules,
+			},
+		}
+	}
+
+	tcs := []struct {
+		name              string
+		cluster           *scyllav1.ScyllaCluster
+		services          map[string]*corev1.Service
+		expectedIngresses []*networkingv1.Ingress
+	}{
+		{
+			name:              "no ingresses when cluster isn't exposed",
+			cluster:           basicScyllaCluster,
+			services:          map[string]*corev1.Service{},
+			expectedIngresses: nil,
+		},
+		{
+			name: "no ingresses when ingresses are explicitly disabled",
+			cluster: func() *scyllav1.ScyllaCluster {
+				cluster := basicScyllaCluster.DeepCopy()
+				cluster.Spec.DNSDomains = []string{"public.scylladb.com", "private.scylladb.com"}
+				cluster.Spec.ExposeOptions = &scyllav1.ExposeOptions{
+					CQL: &scyllav1.CQLExposeOptions{
+						Ingress: &scyllav1.IngressOptions{
+							Disabled:         pointer.Bool(true),
+							IngressClassName: "cql-ingress-class",
+							Annotations: map[string]string{
+								"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+							},
+						},
+					},
+				}
+
+				return cluster
+			}(),
+			services:          map[string]*corev1.Service{},
+			expectedIngresses: nil,
+		},
+		{
+			name: "ingress objects are generated for every domain",
+			services: map[string]*corev1.Service{
+				"any":    newIdentityService("any"),
+				"node-1": newMemberService("node-1", "host-id-1"),
+				"node-2": newMemberService("node-2", "host-id-2"),
+				"node-3": newMemberService("node-3", "host-id-3"),
+			},
+			cluster: func() *scyllav1.ScyllaCluster {
+				cluster := basicScyllaCluster.DeepCopy()
+				cluster.Spec.DNSDomains = []string{"public.scylladb.com", "private.scylladb.com"}
+				cluster.Spec.ExposeOptions = &scyllav1.ExposeOptions{
+					CQL: &scyllav1.CQLExposeOptions{
+						Ingress: &scyllav1.IngressOptions{
+							IngressClassName: "cql-ingress-class",
+							Annotations: map[string]string{
+								"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+							},
+						},
+					},
+				}
+
+				return cluster
+			}(),
+			expectedIngresses: []*networkingv1.Ingress{
+				newIngress("any-cql",
```

Comment:
nit: these expected values are hard to asses whether defined correctly. I know it's a bit painful but I'd inline the expected values

Reply:
expanded